### PR TITLE
Hints should support content/bodies

### PR DESCRIPTION
**Pull request by jsight**
https://issues.jboss.org/browse/WINDUP-497

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/457/
Test PASSed.

File: config/api/src/main/java/org/jboss/windup/config/RuleSubset.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ public void perform(Rewrite rewrite, EvaluationContext context)
                 subContext.setState(RewriteState.EVALUATING);
                 subContext.put(Rule.class, rule);
 
-                event.selectionPush();
```
Comment (lincolnthree) [inline config/api/src/main/java/org/jboss/windup/config/RuleSubset.java]: Good.

File: reporting/api/src/main/java/org/jboss/windup/reporting/config/HintBuilderIn.java [hunk 2]
```diff
@@ -15,6 +15,14 @@ public HintBuilderIn(String fileVariable)
     }
 
     /**
+     * Sets the title of the {@link Hint}.
+     */
+    public HintBuilderTitle titled(String title)
+    {
+        return new HintBuilderTitle(title);
```
Comment (lincolnthree) [inline reporting/api/src/main/java/org/jboss/windup/reporting/config/HintBuilderIn.java]: Good.

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/InlineHintModel.java [hunk 3]
```diff
@@ -17,69 +17,82 @@
...
+     * A short descriptive text describing the problem covered by this hint
+     */
+    @Property(TITLE)
+    String getTitle();
+
+    /**
+     * Set the text to be displayed within this {@link InlineHintModel} in the designated {@link FileModel}.
+     */
+    @Property(HINT)
+    void setHint(String hint);
```
Comment (lincolnthree) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/InlineHintModel.java]: Good.

File: reporting/impl/src/main/java/org/jboss/windup/reporting/xml/HintHandler.java [hunk 4]
```diff
@@ -30,9 +30,9 @@
  * Alternatively, the hint can be in its own element. This is primary useful for longer hint content:
  * 
  * <pre>
- * &lt;hint effort="8"&gt;
+ * &lt;hint title="Short description" effort="8"&gt;
  *  &lt;message&gt;
- *          Longer help contents go here
+ *          Longer help contents go here (markdown format is supported)
```
Comment (lincolnthree) [inline reporting/impl/src/main/java/org/jboss/windup/reporting/xml/HintHandler.java]: Excellent.

**Pull request merged**